src/coverlet.core/Helpers/InstrumentationHelper.cs

```diff
@@ -76,34 +78,86 @@ public static void DeleteHitsFile(string path)
             // Retry hitting the hits file - retry up to 10 times, since the file could be locked
             // See: https://github.com/tonerdo/coverlet/issues/25
             var retryStrategy = CreateRetryStrategy();
-
             RetryHelper.Retry(() => File.Delete(path), retryStrategy, 10);
         }
 
-        public static IEnumerable<string> GetExcludedFiles(IEnumerable<string> excludeRules,
-                                                           string parentDir = null)
+        public static bool IsModuleExcluded(string module, string[] filters)
+        {
+            if (filters == null)
+                return false;
+
+            bool isMatch = false;
+            module = Path.GetFileNameWithoutExtension(module);
+
+            foreach (var filter in filters)
+            {
+                if (!IsValidFilterExpression(filter))
+                    continue;
+
+                string modulePattern = filter.Substring(1, filter.IndexOf(']') - 1);
+                string typePattern = filter.Substring(filter.IndexOf(']') + 1);
+
+                modulePattern = WildcardToRegex(modulePattern);
+
+                var regex = new Regex(modulePattern);
+                isMatch = regex.IsMatch(module) && typePattern == "*";
+            }
+
+            return isMatch;
+        }
+
+        public static bool IsTypeExcluded(string module, string type, string[] filters)
+        {
+            if (filters == null)
+                return false;
+
+            bool isMatch = false;
+            module = Path.GetFileNameWithoutExtension(module);
+
+            foreach (var filter in filters)
+            {
+                if (!IsValidFilterExpression(filter))
+                    continue;
+
+                string typePattern = filter.Substring(filter.IndexOf(']') + 1);
+                string modulePattern = filter.Substring(1, filter.IndexOf(']') - 1);
+
+                typePattern = WildcardToRegex(typePattern);
+                modulePattern = WildcardToRegex(modulePattern);
+
+                isMatch = new Regex(typePattern).IsMatch(type) && new Regex(modulePattern).IsMatch(module);
```

Comment:
For each module, this method is called. For each filter these two regexes are created. This can be cached in some sort of object and the regex can be created using compile mode to speed up the process and to reduce GC presure